### PR TITLE
Add a Dockerfile

### PR DESCRIPTION
**Pull request by PunKeel**
This is not a perfect Dockerfile, but is a perfect base for anyone willing to run PrivateBin
inside a container. There are a lot of ways to improve it, first being "having configuration
in environment variables".
Anyway, this Dockerfile is great as a basis, and it is possible to "fork" the image and add
our own config file into it.
Another improvement might be "adding a docker-compose.yml full stack, with mysql+bin".

Feel free to do it \o/

* [x] I agree to release the changes in this PR under the Zlib/libpng license as shown in the  [LICENSE file](https://github.com/PrivateBin/PrivateBin/blob/master/LICENSE.md#zliblibpng-license-for-privatebin).

* [x] To the extent possible under law, I have waived all copyright and related or neighboring rights to this PR and publish it as public domain.

## Changes
* Add a Dockerfile and a dockerignore (to prevent information leak)

## To do - Won't do
* [ ] Docker Compose
* [ ] Add environment variables for configuration
* [ ] Enable (or not) the .htaccess
* [x] Composer install? Depends on #150

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Documentation, might leak version number
```
Comment (rugk) [inline .dockerignore]: Actually the version number is leaked in PrivateBin itself (e.g. on the bottom of the site it is shown and when loading some JS files it is appended).

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN apt-get update && apt-get install -y \
+        libfreetype6-dev \
+        libjpeg62-turbo-dev \
+        libpng12-dev \
+        wget \
+        zip \
+        unzip; \
+    # We install and enable php-gd
+    docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/; \
```
Comment (rugk) [inline Dockerfile]: You might also want to install mycrypt (or Libsodium) for the random number generator. But if read access to /dev/urandom is guaranteed, then this is of course not necessary.

**Comment by rugk** [comment]
Additional there is already an [unofficial Docker file](https://github.com/Wonderfall/dockerfiles/tree/master/privatebin), whcih is bsaed on nginx & PHP7. See https://github.com/PrivateBin/PrivateBin/issues/99

File: Dockerfile (hunk 2 above)
Comment (PunKeel) [inline Dockerfile]: Yes /dev/uraneom is available

File: .dockerignore (hunk 1 above)
Comment (PunKeel) [inline .dockerignore]: That's up to you to keep or remove these lines, then :)

Comment (rugk) [inline .dockerignore]: I think this exclusions are still useful, but they rather belong to the "unnecessary files" section. 😄

**Comment by elrido** [comment]
To clarify: This dockerfile uses a [Debian Jessie container and compiles the PHP 7.1 sources](https://hub.docker.com/_/php/), while the other project uses an [Alpine Linux container and the PHP 7.x packages of that distro](https://github.com/Wonderfall/dockerfiles/blob/master/privatebin/Dockerfile).

**Pull request merged**